app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/SensorSendQueue.java

```diff
@@ -22,23 +23,18 @@
     @Column(name = "success", index = true)
     public boolean success;
 
-
-    public static SensorSendQueue nextSensorJob() {
-        SensorSendQueue job = new Select()
-                .from(SensorSendQueue.class)
-                .where("success =", false)
-                .orderBy("_ID desc")
-                .limit(1)
-                .executeSingle();
-        return job;
-    }
-
-    public static List<SensorSendQueue> queue() {
-        return new Select()
+    
+    public static List<SensorSendQueue> mongoQueue(boolean xDripViewerMode) {
+        List<SensorSendQueue> values = new Select()
                 .from(SensorSendQueue.class)
-                .where("success = ?", false)
                 .orderBy("_ID desc")
+                .limit(xDripViewerMode ? 100 : 0)
                 .execute();
+        if (xDripViewerMode) {
+             java.util.Collections.reverse(values);
```

Comment:
If we just upload in xDripViewerMode, the order could be set in the query.
But from my standpoind we could very well upload the data in any case. A new Sensor is set every couple of days; maybe when the battery decays we would need some more updates but is still not very much and would not overload any connection.